collector/info_schema_innodb_cmp.go

```diff
@@ -0,0 +1,96 @@
+// Scrape `information_schema.INNODB_CMP`.
+
+package collector
+
+import (
+	"database/sql"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const innodbCmpQuery = `
+		SELECT 
+		  page_size, compress_ops, compress_ops_ok, compress_time, uncompress_ops, uncompress_time 
+		  FROM information_schema.innodb_cmp
+		`
+
+var (
+	infoSchemaInnodbCmpCompressOps = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, informationSchema, "innodb_cmp_compress_ops_total"),
+		"Number of times a B-tree page of the size PAGE_SIZE has been compressed.",
+		[]string{"page_size"}, nil,
+	)
+	infoSchemaInnodbCmpCompressOpsOk = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, informationSchema, "innodb_cmp_compress_ops_ok_total"),
+		"Number of times a B-tree page of the size PAGE_SIZE has been successfully compressed.",
+		[]string{"page_size"}, nil,
+	)
+	infoSchemaInnodbCmpCompressTime = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, informationSchema, "innodb_cmp_compress_time_seconds_total"),
+		"Total time in seconds spent in attempts to compress B-tree pages.",
+		[]string{"page_size"}, nil,
+	)
+	infoSchemaInnodbCmpUncompressOps = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, informationSchema, "innodb_cmp_uncompress_ops_total"),
+		"Number of times a B-tree page of the size PAGE_SIZE has been uncompressed.",
+		[]string{"page_size"}, nil,
+	)
+	infoSchemaInnodbCmpUncompressTime = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, informationSchema, "innodb_cmp_uncompress_time_seconds_total"),
+		"Total time in seconds spent in uncompressing B-tree pages.",
+		[]string{"page_size"}, nil,
+	)
+)
+
+// ScrapeInnodbCmp collects from `information_schema.innodb_cmp`.
+func ScrapeInnodbCmp(db *sql.DB, ch chan<- prometheus.Metric) error {
+
+	informationSchemaInnodbCmpRows, err := db.Query(innodbCmpQuery)
+	if err != nil {
+		return err
+	}
+	defer informationSchemaInnodbCmpRows.Close()
+
+	// The client column is assumed to be column[0], while all other data is assumed to be coerceable to float64.
```

Comment:
I think this documentation is out of date with the refactoring.  Part of the reason I prefer the [explicit mapping](https://github.com/prometheus/mysqld_exporter/blob/master/collector/info_schema_innodb_metrics.go#L65-L67) rather than automatic mapping of columns is that there's less need for documentation like the code makes it clear what columns are mapped to what values.

Reply:
Mapping parts were rewritten.